urllib3/util/retry.py

```diff
@@ -330,7 +338,7 @@ def increment(self, method=None, url=None, response=None, error=None,
 
         elif error and self._is_read_error(error):
             # Read retry?
-            if read is False:
+            if read is False or (method is not None and not self._is_method_retryable(method)):
```

Comment:
@Lukasa I don't think it's actually `None` anywhere (except some tests), but the default value for the argument is None (which I assumed meant it was a valid value to pass) so I added that check.